src/platform/common/platform/types.node.ts

```diff
@@ -12,24 +12,8 @@ export type WriteStream = fs.WriteStream;
 
 export const IFileSystemNode = Symbol('IFileSystemNode');
 export interface IFileSystemNode extends IFileSystem {
-    areLocalPathsSame(path1: string, path2: string): boolean;
-    createLocalDirectory(path: string): Promise<void>;
-    /**
-     * Copies a local file/directory.
-     * @param {{ overwrite: boolean }} [options] Defaults to overwriting existing files/directories.
-     */
-    copyLocal(source: string, destination: string, options?: { overwrite: boolean }): Promise<void>;
-    deleteLocalFile(path: string): Promise<void>;
```

Comment:
Thats a lot of unnecessary API that callers don't need to worry about.
In the custom implemetnation we check the scheme of the Uri and use the appropriate methods (i.e. when dealing with local files we use fs module else revert to using VS Code API)

Hence a slimmer interface and simpler to use.
Also optimized the code to always use fs modules instead of VS Code when dealing with local fs resources